source/iopipe/json/serialize.d

```diff
@@ -65,6 +77,17 @@ private void onField(P, JT, T, size_t N)(ref P policy, ref JT tokenizer, ref T i
     static assert(N == SerializableMembers!T.length);
     alias members = SerializableMembers!T;
     alias ignoredMembers = AllIgnoredMembers!T;
+
+    static foreach(idx, m; members)
+    {
+        static if(hasUDA!(__traits(getMember, T, m), extras))
+        {
+            // this is the extras member, it holds any extra data that was not
+            // specified as a member.
+            static assert(is(typeof(__traits(getMember, T, m)) == JSONValue!S, S));
+            enum extrasMember = m;
+        }
+    }
```

Comment:
I realize that this is somewhat copying what I did.

There are a couple issues here. First, you are looping twice. And you can do this inside the loop below. Second issue is that the loop below doesn't exclude the extras member from the list of serializable members (the original code does).

What I'd suggest is to remove this loop, and inside the switch loop:

```d
static if(hasUDA!(__traits(getMember, T, memberName), extras)) {
    enum extrasMember = memberName;
} else {
    ... // normal deserialize code as you have it
}
```

Reply:
Just realized, this loop has to be here, because extrasMember defined inside the other loop would be inside the switch scope.